### ISSUE TITLE
Unable to source ROS source files (creates temporary files in /tmp/)

### ISSUE THREAD

**Issue by routeaccess** [sorin-ionescu/prezto#795]
When I'm trying to use the [ROS](http://www.ros.org/) framework for robotics, I'm not able to source the overlay files. The command `source /opt/ros/indigo/setup.zsh` fails with the following error 

```bash
/opt/ros/indigo/setup.sh:52: file exists: /tmp/setup.sh.CkuqltdAon
Failed to run '"/opt/ros/indigo/_setup_util.py" ': return code 1
```

The curious fact is, the command when added to `.zshenv` works fine, but throws the above error when it is run from the terminal or when added to `.zshrc` (The error output shows up at the beginning of a new terminal when added to `.zshrc`)

This is the code snippet which fails (lines 41-52 from `/opt/ros/indigo/setup.sh` which is called by `/opt/ros/indigo/setup.zsh`) :
```bash
# remember type of shell if not already set
if [ -z "$CATKIN_SHELL" ]; then
  CATKIN_SHELL=sh
fi

# invoke Python script to generate necessary exports of environment variables
_SETUP_TMP=`mktemp /tmp/setup.sh.XXXXXXXXXX`
if [ $? -ne 0 -o ! -f "$_SETUP_TMP" ]; then
  echo "Could not create temporary file: $_SETUP_TMP"
  return 1
fi
CATKIN_SHELL=$CATKIN_SHELL "$_SETUP_UTIL" $@ > $_SETUP_TMP
```
This snippet fails at the last line.

The file mentioned in the error inside `/tmp/` is only a temporary file made with a random string name appended to it and the script being sourced runs fine when used in `.zshenv` which leads me to believe it is a bug. Ideally I should be able to run it inside the terminal as is possible when not using a framework.

**Comment by sorin-ionescu** [comment]
[Line 22](https://github.com/sorin-ionescu/prezto/blob/master/modules/directory/init.zsh#L22) of *modules/direcotory/init.zsh* is your problem.

I have unset that option because I have got burned overwriting the wrong files with no backups in the past. You may delete that line, add `setopt clobber` in *zshrc* bellow *Customize to your needs...* or temporarily `setopt clobber` then `unsetopt clobber` around the sourcing of your file.

**Comment by routeaccess** [comment]
Thank you for the quick resolution - the issue has changed now. Your suggestion took care of the previous error, but this is the error it throws now:
```bash
setup.sh:88: parse error near `\n'
```
I've checked the file in question and this appears to be the last line in the file, _which is blank_. The file is valid since it works fine when sourced from `.zshenv`, so this must not be an error due to a missing brace or such.

**Comment by sorin-ionescu** [comment]
That code may be BASH-only and will not work on Zsh.

**Comment by routeaccess** [comment]
`setup.sh` is a file that is internally sourced by `setup.zsh` (which is the file I'm trying to source - please refer to issue details on top for more details ) after setting a few variables in place for `zsh` compatibility, so it should work in `zsh`.

Again, it works perfectly fine when `setup.zsh` is sourced from within `.zshenv` (as mentioned up top) so I hope there is some way to get around this. I can gist the files if so required.

**Comment by sorin-ionescu** [comment]
You shouldn't be sourcing it in zshenv but zprofile.

Syntax errors are not a Prezto-induced problem since it would not work with or without ROS.

Without seeing those ROS files, I have no way of figuring out what the problem is.

**Comment by routeaccess** [comment]
Please find the files `setup.zsh`, `setup.sh` and `_setup_util.py` [here](https://gist.github.com/routeaccess/e9299bd32969c4c3e825).

- These files are autogenerated by `ROS`,  not hand-written / edited.
- There isn't a syntax error in them since sourcing the same files work fine when done from `zprofile` or `zshenv` ( or when not using Prezto).
- All three files gisted at the above link are inside `$_CATKIN_SETUP_DIR`.

I sincerely thank you for your patience, I hope you take your time and get to the bottom of this at your convenience.

**Comment by sorin-ionescu** [comment]
After I `setopt clobber`, it sources and executes fine on my end. I can't help more.

**Comment by routeaccess** [comment]
For anyone stumbling by this issue in future, the problem was caused by this custom alias I had set
```bash
alias done="exit"
```
 forgetting that `done` was a valid shell keyword which the alias may overwrite. Serves me right for only checking if the term was an executable/program before setting the alias. Needless to say, renaming the alias solved the parse error.

@sorin-ionescu :  Only once I got over the first error as per your suggestion, was this possible to debug on my own. Thank you for the quick response.
